packages/core/utils/src/convert-query-params.ts

```diff
@@ -357,42 +366,29 @@ const createTransformer = ({ getModel }: TransformerOptions) => {
       const isAllowedAttributeForFragmentPopulate =
         isDynamicZoneAttribute(attribute) || isMorphToRelationalAttribute(attribute);
 
-      if (isAllowedAttributeForFragmentPopulate && hasFragmentPopulateDefined(subPopulate)) {
-        return {
-          ...acc,
-          [key]: {
-            on: Object.entries(subPopulate.on).reduce(
-              (acc, [type, typeSubPopulate]) => ({
-                ...acc,
-                [type]: convertNestedPopulate(typeSubPopulate, getModel(type)),
-              }),
-              {}
-            ),
-          },
-        };
-      }
-
-      // TODO: This is a query's populate fallback for DynamicZone and is kept for legacy purpose.
-      //       Removing it could break existing user queries but it should be removed in V5.
-      if (isDynamicZoneAttribute(attribute)) {
-        const populates = attribute.components
-          .map((uid) => getModel(uid))
-          .map((schema) => convertNestedPopulate(subPopulate, schema))
-          .map((populate) => (populate === true ? {} : populate)) // cast boolean to empty object to avoid merging issues
-          .filter((populate) => populate !== false);
-
-        if (isEmpty(populates)) {
-          return acc;
+      if (isAllowedAttributeForFragmentPopulate) {
+        if (hasFragmentPopulateDefined(subPopulate)) {
```

Comment:
Currently the hasFragmentPopulateDefined function only verifies that an on property is defined and non nil, but it doesn't make sure no other props are present. Might be worth adding a check to the function directly?

Reply:
not sure I understood, you mean it doesn't check if it has any properties ?

Reply:
Basically, we need to check that only `on` is defined in the sub populate, no other key should be accepted